File: Leetcode/238-Product-of-Array-Except-Self/ProductOfArrayExceptSelf.go
func productExceptSelf(nums []int) []int {
	res := make([]int, len(nums))
	p := 1
	res[0] = 1
	for i := 1; i < len(nums); i++ {
		p *= nums[i-1]
		res[i] = p
	}
	p = 1
	for i := len(nums) - 2; i >= 0; i-- {
		p *= nums[i+1]
		res[i] *= p
	}
	return res
}

File: Leetcode/238-Product-of-Array-Except-Self/ProductOfArrayExceptSelf.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    var res = []
    var p=1
    res.push(1)
    for (var i=1; i<nums.length; i++) {
        p*=nums[i-1]
        res.push(p)
    }

    p=1
    for (var i=nums.length-2; i>=0; i--) {
        p*=nums[i+1]
        res[i]*=p
    }
    return res
};
